rclpy/rclpy/node.py

```diff
@@ -1652,3 +1652,68 @@ def assert_liveliness(self) -> None:
         """
         with self.handle as capsule:
             _rclpy.rclpy_assert_liveliness(capsule)
+
+    def _get_info_by_topic(
+        self,
+        topic_name: str,
+        no_mangle: bool,
+        func: Callable[[object, str, bool], List[Dict]]
+    ) -> List[Dict]:
+        fq_topic_name = expand_topic_name(topic_name, self.get_name(), self.get_namespace())
+        validate_topic_name(fq_topic_name)
+        with self.handle as node_capsule:
+            return func(node_capsule, fq_topic_name, no_mangle)
+
+    def get_publishers_info_by_topic(
+            self, topic_name: str, no_mangle: bool = False) -> List[Dict]:
+        """
+        Return a list of publishers publishing to a given topic.
+
+        The returned parameter is a list of dictionaries, where each dictionary will
+        contain the node name, node namespace, topic type, participant's GID and its QoS profile.
+
+        When the `no_mangle` parameter is `true`, the provided `topic_name` should be a valid topic
+        name for the middleware (useful when combining ROS with native middleware (e.g. DDS) apps).
+        When the `no_mangle` parameter is `false`, the provided `topic_name` should follow
+        ROS topic name conventions.
+
+        `topic_name` may be a relative, private, or fully qualified topic name.
+        A relative or private topic will be expanded using this node's namespace and name.
+        The queried `topic_name` is not remapped.
+
+        :param topic_name: the topic_name on which to find the publishers.
+        :param no_mangle: no_mangle if `true`, `topic_name` needs to be a valid middleware topic
+                          name, otherwise it should be a valid ROS topic name. Defaults to `false`.
+        :return: a list of dictionaries representing all the publishers on this topic.
+        """
+        return self._get_info_by_topic(
+            topic_name,
+            no_mangle,
+            _rclpy.rclpy_get_publishers_info_by_topic)
+
+    def get_subscriptions_info_by_topic(
+            self, topic_name: str, no_mangle: bool = False) -> List[Dict]:
+        """
+        Return a list of subscriptions to a given topic.
+
+        The returned parameter is a list of dictionaries, where each dictionary will contain
+        the node name, node namespace, topic type, participant's GID and its QoS profile.
```

Comment:
Same comment about use of `participant` here.

Reply:
Switched over to using `topic endpoint`.